staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -214,6 +215,9 @@ func (le *LeaderElector) acquire(ctx context.Context) bool {
 		klog.Infof("successfully acquired lease %v", desc)
 		cancel()
 	}, le.config.RetryPeriod, JitterFactor, true, ctx.Done())
+	if succeeded {
+		leaderGauge.WithLabelValues(le.config.Name).Set(1.0)
```

Comment:
any reason not do do this on 214 where we record the event?

Reply:
No good reason. Making the code consistent to record the gauge value after recording the event.